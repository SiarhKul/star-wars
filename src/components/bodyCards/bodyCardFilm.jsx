import React from "react";

export const BodyCardFilm = ({ card: { title, director, release_date } }) => {
	return (
		<div className="body">
			<div className="body__text">Name:{title}</div>
			<div className="body__text">Director:{director}</div>
			<div className="body__text">Release date:{release_date}</div>
		</div>
	);
};
