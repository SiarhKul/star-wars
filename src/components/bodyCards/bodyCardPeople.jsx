import React from "react";

export const BodyCardPeople = ({ card }) => (
		<div className="body">
			<div className="body__text">Name:{card.name}</div>
			<div className="body__text">Birth year:{card?.birth_year}</div>
			<div className="body__text">Gender:{card?.gender}</div>
		</div>
	);
