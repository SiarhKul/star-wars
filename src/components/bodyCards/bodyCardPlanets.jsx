import React from "react";

export const bodyCardPlanets = ({ card }) => {
	return (
		<div className="body">
			<div className="body__text">Name:{card.name}</div>
			<div className="body__text">Population:{card.population}</div>
		</div>
	);
};
