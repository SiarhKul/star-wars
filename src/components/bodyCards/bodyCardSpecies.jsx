import React from "react";

export const BodyCardSpecies = ({ card }) => (
		<div className="body">
			<div className="body__text">Name:{card.name}</div>
			<div className="body__text">Language:{card.language}</div>
		</div>
	);
