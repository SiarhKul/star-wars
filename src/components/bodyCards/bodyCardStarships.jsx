import React from "react";

export const BodyCardStarships = ({ card }) => {
	return (
		<div className="body">
			<div className="body__text">Name:{card.name}</div>
			<div className="body__text">Cost:{card.cost_in_credits}</div>
			<div className="body__text">Passangers:{card.passengers}</div>
		</div>
	);
};
