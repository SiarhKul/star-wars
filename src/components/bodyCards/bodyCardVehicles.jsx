import React from "react";

export const BodyCardVehicles = ({ card }) => (
		<div className="body">
			<div className="body__text">Name:{card?.name}</div>
			<div className="body__text">Model:{card?.model}</div>
			<div className="body__text">Crew:{card?.crew}</div>
		</div>
	);
