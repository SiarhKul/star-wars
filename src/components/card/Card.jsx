import React, { useCallback } from "react";
import { getAbbreviation } from "../../utils";

export const Card = React.memo(
	({ uniqueName, card, BodyComponent, onUniqueQueryParam }) => {
		const setUniqueQueryParam = useCallback(() => {
			onUniqueQueryParam(card);
		}, []);

		
		return (
			<div className="card-wrapper">
				<div className="card" onClick={setUniqueQueryParam}>
					<div className="avatar">
						<div className="avatar__abbr">{getAbbreviation(uniqueName)}</div>
					</div>
					<BodyComponent card={card} />
				</div>
			</div>
		);
    }
);
