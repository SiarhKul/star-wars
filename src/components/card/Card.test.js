import { shallow } from 'enzyme';
import React from 'react';
import { BodyCardVehicles } from '../bodyCards/bodyCardVehicles';
import { Card } from './Card';

describe('Test <Card/>', () => {
  const mockedProps = {
    uniqueName: 'Sail bare',
    card: {},
    BodyComponent: BodyCardVehicles,
    onUniqueQueryParam: jest.fn(),
  };
  const component = shallow(<Card {...mockedProps} />);

  it('render snapshot <Card/>', () => {
    expect(component).toMatchSnapshot();
  });

  it('should do click on card', () => {
    const card = component.find('.card');
    expect(mockedProps.onUniqueQueryParam.mock.calls.length).toBe(0);
    card.simulate('click');
    expect(mockedProps.onUniqueQueryParam.mock.calls.length).toBe(1);
  });
});
