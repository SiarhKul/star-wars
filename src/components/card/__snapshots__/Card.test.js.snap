// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Card/> render snapshot <Card/> 1`] = `
<div
  className="card-wrapper"
>
  <div
    className="card"
    onClick={[Function]}
  >
    <div
      className="avatar"
    >
      <div
        className="avatar__abbr"
      >
        SS
      </div>
    </div>
    <BodyCardVehicles
      card={Object {}}
    />
  </div>
</div>
`;
